compose/ui/ui-test/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skiko.kt

```diff
@@ -205,30 +223,45 @@ open class SkikoComposeUiTest @InternalTestApi constructor(
     private val testOwner = SkikoTestOwner()
     private val testContext = TestContext(testOwner)
 
-    fun <R> runTest(block: SkikoComposeUiTest.() -> R): R {
-        return composeRootRegistry.withRegistry {
-            withScene {
+    fun runTest(
+        block: suspend SkikoComposeUiTest.() -> Unit
+    ): TestResult {
+        composeRootRegistry.setupRegistry()
+        scene = runOnUiThread(::createUi)
```

Comment:
Some explanation:

Creating the scene manually instead of using `withScene {`. 
We need an `effectContext` of the recomposer to be passed to the runTestContext, so with the current structure of classes, the scene must be created before we run `kotlinx.coroutines.test.runTest(`. See `combinedCoroutineContext`